test/random-battles/all-gens.js

```diff
@@ -17,10 +17,13 @@ describe('value rule support (slow)', () => {
 
 		testTeam({format: 'gen8multirandombattle', rounds: 100}, team => assert.equal(team.length, 3));
 		testTeam({format: 'gen8cap1v1', rounds: 100}, team => assert.equal(team.length, 3));
+
+		testTeam({format: 'gen7randombattle', rounds: 100}, team => assert.equal(team.length, 6));
 	});
 
 	// TODO: Support gen 9 set format
 	for (let gen = 1; gen <= 8; gen++) {
+		if (gen === 7) continue;
```

Comment:
Why are we exempting Gen 7 from these tests? Does this revamp not support Max Move Count?

Reply:
Due to the frontloading of moveset generation, it doesn't support Max Move Count if it is set lower than 4, as some sets have more than one 'enforced' move.

Reply:
Then why not add a generation-specific check to skip testing only those `Max Move Count`s lower than 4 in Gen 7, similar to what Gen 1 has? That seems better than bypassing the test altogether.

Reply:
I've adjusted the Max Move Count tests to include gen 7 and gen 9